### ISSUE TITLE
Microsoft.DotNet.ApiCompat

### ISSUE THREAD

**Issue by petrsvihlik** [petrsvihlik/WopiHost#153]
- https://www.nuget.org/packages?q=Microsoft.DotNet.ApiCompat
- https://github.com/dotnet/arcade/tree/main/src/Microsoft.DotNet.ApiCompat

**Comment by petrsvihlik** [comment]
## Plan: API-compat checks in PR pipeline + auto-bump baseline at release

### Current state
- `Directory.Build.props` already has `EnablePackageValidation=true` with `PackageValidationBaselineVersion=4.0.0` — but:
  - The PR workflow only runs `dotnet build` + `dotnet test`, never `pack`, so **no API-compat check actually runs on PRs today**.
  - The pinned baseline `4.0.0` is two majors stale (latest published is `5.1.0`), so even the pack-time check at release is comparing against an irrelevant surface.
- Three csproj files (`Abstractions`, `Core`, `Discovery`) plus `Directory.Packages.props` reference the obsolete preview `Microsoft.DotNet.PackageValidation` PackageReference. That validator has been built into the SDK since 6.0.100; the explicit reference is dead weight.
- `Microsoft.DotNet.ApiCompat.Tool` (the standalone global tool referenced in this issue) is the right pick for the PR-comment use case: it can compare a freshly-packed `nupkg` against any baseline `nupkg` and emit `CP*`/`PKV*` diagnostic IDs, outside of `dotnet pack`.

### Scope of this work

#### 1. New job in `.github/workflows/pull_request.yml`: `api-compat`
- Gated by the existing `Approve` job (so fork PRs still require human approval before any code runs).
- Steps: `dotnet pack -c Release` → for each of the 6 packable libraries (`Abstractions`, `Core`, `Discovery`, `Url`, `FileSystemProvider`, `MemoryLockProvider`), look up the latest stable on nuget.org via the v3 flat-container index, download that nupkg, run `dotnet apicompat package --package <new> --baseline-package <baseline>`.
- **Non-blocking** — collects findings, never fails the job (this repo cuts intentional breaks at major versions; we don't want PRs red-flagged for those).
- Aggregates into a single markdown report (one `<details>` block per package, ✅ if clean, code-block of CP/PKV findings if not) and posts via `marocchino/sticky-pull-request-comment@v2` with a header marker so the comment updates in place on each push.
- Cobalt is `IsPackable=false` and not on nuget.org → excluded.

#### 2. Cleanup (same PR)
- Bump `PackageValidationBaselineVersion` 4.0.0 → 5.1.0 in `Directory.Build.props`.
- Remove `<PackageReference Include=\"Microsoft.DotNet.PackageValidation\" />` from `WopiHost.Abstractions`, `WopiHost.Core`, `WopiHost.Discovery` csproj.
- Drop the `PackageVersion` entry from `Directory.Packages.props`.

#### 3. Auto-bump in `.github/workflows/release.yml`
- After the publish-to-nuget step succeeds (and only for non-prerelease, non-draft releases), append a job that:
  - Updates `Directory.Build.ps` `PackageValidationBaselineVersion` to the just-published version.
  - Opens a PR via `peter-evans/create-pull-request` titled \"chore: bump PackageValidationBaselineVersion to vX.Y.Z\" against `master`.
- A PR (not direct push) keeps it review-able and respects branch protection.

### Risks / edges
- After a major version cut with intentional breaks, the next round of PRs will be noisy in the sticky comment until the auto-bump PR (from step 3) merges. Acceptable trade-off — the sticky comment is informational, not blocking.
- `dotnet apicompat` exits non-zero on any diff. The PR job must allow non-zero exit so the comment still posts (`continue-on-error` / `|| true`).
- `pull_request_target` security: the existing `Approve` gate + `Integrate Pull Request` environment for forks remain in place. The api-compat job reads ApiCompat output (generated from compiled assemblies, not PR-controlled text), so the comment body is naturally safe from injection.

### Out of scope / follow-ups
- Optional `api-breaking` PR label when CP-class findings are present.
- Strict-mode toggling per package (currently default `--enable-strict-mode-for-compatible-tfms=true`).

PR coming next.
